### PR TITLE
Handling double slashes in the HTTP client and server

### LINKED ISSUE #19722: HTTP client nor server is correctly handling double slashes
IIRC per the spec http://foo.com/bar//baz is same as http://foo.com/bar/baz.

This is not handled properly right now .. dispatching fails.

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Handling double slashes at HTTP client and server.
Fixes #19722

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/http/src/main/java/org/ballerinalang/net/http/HttpConstants.java [hunk 1]
```diff
@@ -431,6 +431,8 @@
 
     public static final String COLON = ":";
     public static final String DOLLAR = "$";
+    public static final String DOWBLE_SLASH = "//";
```
Comment (Bhashinee) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/HttpConstants.java]:
```suggestion
    public static final String DOUBLE_SLASH = "//";
```

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/44_http_url.bal [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
+service urlClientTest on new http:Listener(9255)  {
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "//"
+    }
+    resource function hello(http:Caller caller, http:Request request) {
+        string value = "";
+        var response1 = urlClient->get("//test");
```
Comment (chamil321) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/44_http_url.bal]: Shouldn't we also consider multiple slashes normalization too???
"/test/////go"

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/44_http_url.bal [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        if (response1 is http:Response) {
+            var result = response1.getTextPayload();
+            if (result is string) {
+                value = result;
+            } else {
+                error err = result;
+                value = err.reason();
```
Comment (Bhashinee) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/44_http_url.bal]: Do we need to have this 'err' ? We can simply do something like this?
```suggestion
                value = result.reason();
```

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/44_http_url.bal [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    resource function hello(http:Caller caller, http:Request request) {
+        string value = "";
+        var response1 = urlClient->get("//test");
+        if (response1 is http:Response) {
+            var result = response1.getTextPayload();
+            if (result is string) {
+                value = result;
+            } else {
+                error err = result;
```
Comment (Bhashinee) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/44_http_url.bal]: This line is unnecessary. Shall we remove that?

File: stdlib/http/src/main/java/org/ballerinalang/net/http/HttpResource.java [hunk 5]
```diff
@@ -72,6 +73,7 @@
 
     private boolean transactionAnnotated = false;
 
+    private static PrintStream console = System.out;
```
Comment (chamil321) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/HttpResource.java]: Is this required?

**Pull request merged**